src/main/java/org/folio/circulation/domain/LoanCheckInService.java

```diff
@@ -16,4 +16,13 @@
       : loan.checkIn(request.getCheckInDate(), systemDateTime,
         request.getServicePointId()));
   }
+
+  public boolean isInHouseUse(Item item, RequestQueue requestQueue,
+    CheckInByBarcodeRequest checkInByBarcodeRequest) {
+
+    return item.isAvailable()
+      && requestQueue.size() == 0
```

Comment:
Consider introducing an `isEmpty` method for this, instead of using the size, it might express the intent better